kratos/includes/mortar_classes.h

```diff
@@ -1319,20 +1319,24 @@ class DualLagrangeMultiplierOperators
      */
     boost::numeric::ublas::bounded_matrix<double, TNumNodes, TNumNodes> CalculateAe()
     {        
+        const double tolerance = std::numeric_limits<double>::epsilon();
+        
         // We compute the norm
         const double norm_me = norm_frobenius(Me);
         
         // Now we normalize the matrix
         const boost::numeric::ublas::bounded_matrix<double, TNumNodes, TNumNodes> normalized_me = Me/norm_me;
         
         // We compute the normalized inverse
-        double aux_det;
-        const boost::numeric::ublas::bounded_matrix<double, TNumNodes, TNumNodes> normalized_inv_me = MathUtils<double>::InvertMatrix<TNumNodes>(normalized_me, aux_det, std::numeric_limits<double>::epsilon()); 
+        double aux_det = MathUtils<double>::DetMat<TNumNodes>(normalized_me);
```

Comment:
you are calculating twice the determinant... once it is done as a part of the inverse calculation. 
As i understand you are afraid it can be 0, however isn't there a way to avoid this?

Reply:
Nope, I think